src/components/manage/Sharing/Sharing.jsx

```diff
@@ -85,183 +80,97 @@ const messages = defineMessages({
   },
 });
 
-/**
- * SharingComponent class.
- * @class SharingComponent
- * @extends Component
- */
-class SharingComponent extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    updateSharing: PropTypes.func.isRequired,
-    getSharing: PropTypes.func.isRequired,
-    updateRequest: PropTypes.shape({
-      loading: PropTypes.bool,
-      loaded: PropTypes.bool,
-    }).isRequired,
-    pathname: PropTypes.string.isRequired,
-    entries: PropTypes.arrayOf(
-      PropTypes.shape({
-        id: PropTypes.string,
-        login: PropTypes.string,
-        roles: PropTypes.object,
-        title: PropTypes.string,
-        type: PropTypes.string,
-      }),
-    ).isRequired,
-    available_roles: PropTypes.arrayOf(PropTypes.object).isRequired,
-    inherit: PropTypes.bool,
-    title: PropTypes.string.isRequired,
-    login: PropTypes.string,
-  };
+const SharingComponent = (props) => {
+  const { title } = useContent();
+  const { login } = useToken();
+  const {
+    entries: props_entries,
+    inherit: props_inherit,
+    available_roles,
+    updateRequest,
+  } = useSharing();
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const [search, setSearch] = useState('');
+  const [inherit, setInherit] = useState(props_inherit);
+  const [isClient, setIsClient] = useState(false);
+  const [entries, setEntries] = useState(props_entries);
+  const pathname = props.location.pathname;
 
-  /**
-   * Default properties
-   * @property {Object} defaultProps Default properties.
-   * @static
-   */
-  static defaultProps = {
-    inherit: null,
-    login: '',
-  };
+  useEffect(() => {
+    dispatch(getSharing(getBaseUrl(pathname), search));
+    setIsClient(true);
+  }, [dispatch, pathname, search, isClient]);
 
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs Sharing
-   */
-  constructor(props) {
-    super(props);
-    this.onCancel = this.onCancel.bind(this);
-    this.onChange = this.onChange.bind(this);
-    this.onChangeSearch = this.onChangeSearch.bind(this);
-    this.onSearch = this.onSearch.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
-    this.onToggleInherit = this.onToggleInherit.bind(this);
-    this.state = {
-      search: '',
-      inherit: props.inherit,
-      entries: props.entries,
-      isClient: false,
-    };
-  }
-
-  /**
-   * Component did mount
-   * @method componentDidMount
-   * @returns {undefined}
-   */
-  componentDidMount() {
-    this.props.getSharing(getBaseUrl(this.props.pathname), this.state.search);
-    this.setState({ isClient: true });
-  }
+  useEffect(() => {
+    dispatch(getSharing(getBaseUrl(pathname), search));
```

Comment:
Beware of lifecycle changes. The toast is only displayed if the request is successfully loaded. we probably don't need any other entry to dep. array other than this.